packages/@storefront/core/src/services/url.ts

```diff
@@ -157,15 +157,40 @@ class UrlService extends BaseService<UrlService.Options> {
 
   filterState(state: Store.State) {
     const { session: { config, ...session }, data, ...rootConfig } = state;
-    const products = this.app.config.history.length === 0 ? [] : data.present.products;
+    let { navigations, products, template } = data.present;
+    let {
+      navigations: autocompleteNavigations,
+      products: autocompleteProducts,
+      template: autocompleteTemplate,
+    } = data.present.autocomplete;
```

Comment:
I think you can actually do this:
```js
    let {
      navigations,
      products,
      template
      autocomplete: {
        navigations: autocompleteNavigations,
        products: autocompleteProducts,
        template: autocompleteTemplate,
      }
    } = data.present;
```

Up to you if you don't think this is clearer.

Reply:
Looks good, will implement.